app/models/user.rb

```diff
@@ -60,17 +59,18 @@ def local_roles
     def global_roles
       GLOBAL_ROLES
     end
-  end
 
-  def self.find_for_authentication(tainted_conditions)
-    subdomain = tainted_conditions[:subdomain]
-    email = tainted_conditions[:email]
+    def find_for_authentication(tainted_conditions)
+      subdomain = tainted_conditions[:subdomain]
+      email = tainted_conditions[:email]
 
-    if subdomain == "config"
-      User.kept.global_administrator.find_first_by_auth_conditions(email:)
-    else
-      local_authority = LocalAuthority.find_by!(subdomain:)
-      local_authority.users.kept.find_first_by_auth_conditions(email:)
+      if subdomain == "config"
+        local_authority_id = nil
+      else
+        local_authority_id = LocalAuthority.find_by!(subdomain:).id
+      end
+
+      kept.find_first_by_auth_conditions({email:}, {local_authority_id:})
```

Comment:
This needs to be two hashes otherwise Devise will drop the `nil` value from the lookup.